Content.Server/Chat/Systems/ChatSystem.Events.cs

```diff
@@ -0,0 +1,206 @@
+using Content.Server.Language;
+using Content.Shared.Chat;
+using Content.Shared.Radio;
+
+namespace Content.Server.Chat.Systems
+{
+    public class EntityChat
+    {
+        /// <summary>
+        /// The origin of the chat message.
+        /// </summary>
+        public EntityUid Source;
+
+        /// <summary>
+        /// The original chat message.
+        /// </summary>
+        public string Message;
+
+        /// <summary>
+        /// A dictionary of recipients to arbitrary data objects.
+        /// </summary>
+        /// <remarks>
+        /// The objects are passed to the recipients at a later stage of processing.
+        /// </remarks>
+        public Dictionary<EntityUid, object> Recipients;
+
+        public ChatChannel Channel;
+
+        /// <summary>
+        /// Which system, if any, has claimed this chat message.
+        /// </summary>
+        /// <remarks>
+        /// If a system claims a chat, they are saying they will be responsible for sending it.
+        /// </remarks>
+        public Type? ClaimedBy;
+
+        /// <summary>
+        /// Arbitrary data that may be attached to a chat message.
+        /// </summary>
+        public object? Data;
+
+        public EntityChat(EntityUid source, string message)
+        {
+            Source = source;
+            Message = message;
+            Recipients = new();
+        }
+    }
+
+    public class EntityChatSpokenData
+    {
+        /// <summary>
+        /// The radio channel that this chat is being transmitted to.
+        /// </summary>
+        public RadioChannelPrototype? RadioChannel;
```

Comment:
The same message should be able to get sent on multiple channels

Reply:
That's a good idea. It shouldn't be too hard to make that an array when I work out the radio listener.